### PR TITLE
Add a check to ensure the Jaeger version is reported properly by the Jaeger Operator

### LINKED ISSUE #1886: Add a check in the E2E tests to ensure #1702 is not repeated
## Requirement - what kind of business use case are you trying to solve?
Add a check to our E2E tests to ensure the version is reported properly. #1702

### PR DESCRIPTION
**Pull request by iblancasa**
## Which problem is this PR solving?
- Resolves #1886

*Note:* blocked by #1885

### CODE REVIEW

File: tests/e2e/upgrade/upgrade/check-jaeger-version.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+
+jaeger=$1
+namespace=$2
+expected_version=$3
+
+check_version
+while [ "$VERSION" != "$expected_version" ]
+do
+    echo "Version missmatch: found $VERSION expected $expected_version"
```
Comment (frzifus) [inline tests/e2e/upgrade/upgrade/check-jaeger-version.sh:20]:
```suggestion
    echo "Version mismatch: found $VERSION expected $expected_version"
```

Comment (iblancasa) [reply tests/e2e/upgrade/upgrade/check-jaeger-version.sh:20]: Good catch!

File: tests/e2e/upgrade/upgrade/check-jaeger-version.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    export VERSION=$(kubectl get jaeger $jaeger -o jsonpath="{.status.version}" -n $namespace)
+}
+
+
+jaeger=$1
+namespace=$2
+expected_version=$3
+
+check_version
+while [ "$VERSION" != "$expected_version" ]
```
Comment (rubenvp8510) [inline tests/e2e/upgrade/upgrade/check-jaeger-version.sh:18]: is this going to while forever if the version doesnt matdh?

**Comment by iblancasa** [comment]
@rubenvp8510 I didn't forget. The problem is I commited something from a GitHub suggestion. This doesn't allow to sign commits yet

File: tests/e2e/upgrade/upgrade/check-jaeger-version.sh (hunk 2 above)
Comment (iblancasa) [reply tests/e2e/upgrade/upgrade/check-jaeger-version.sh:18]: Yes, but KUTTL timeout will finish the execution of this script

**Comment by iblancasa** [comment]
@rubenvp8510, could you take a look to this PR? Thanks!

**Review by rubenvp8510 — APPROVED** [review]

**Pull request merged**